pkgs/by-name/go/goreleaser/package.nix

```diff
@@ -28,6 +28,10 @@ buildGoModule rec {
     "-X main.builtBy=nixpkgs"
   ];
 
+  subPackages = [
```

Comment:
I was curious why we needed [subPackages](https://nixos.org/manual/nixpkgs/unstable/#var-go-subPackages)

Then noticed what happens attempting to build the latest version without `subPackages`:

```
$ nix build .#goreleaser -L
warning: Git tree '/home/rherna/workspace/nixpkgs' is dirty
goreleaser> Running phase: unpackPhase
goreleaser> unpacking source archive /nix/store/zpawjys1wa74q2rg829f1agffkqrxaba-source
goreleaser> source root is source
goreleaser> Running phase: patchPhase
goreleaser> Running phase: updateAutotoolsGnuConfigScriptsPhase
goreleaser> Running phase: configurePhase
goreleaser> Running phase: buildPhase
goreleaser> Building subPackage .
goreleaser> Building subPackage ./cmd
goreleaser> Building subPackage ./dagger
goreleaser> main module (github.com/goreleaser/goreleaser/v2) does not contain package github.com/goreleaser/goreleaser/v2/dagger
error: builder for '/nix/store/57j6mixxkky8kzk517y4a6xjw4p9p44b-goreleaser-2.4.4.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/zpawjys1wa74q2rg829f1agffkqrxaba-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Running phase: buildPhase
       > Building subPackage .
       > Building subPackage ./cmd
       > Building subPackage ./dagger
       > main module (github.com/goreleaser/goreleaser/v2) does not contain package github.com/goreleaser/goreleaser/v2/dagger
       For full logs, run 'nix log /nix/store/57j6mixxkky8kzk517y4a6xjw4p9p44b-goreleaser-2.4.4.drv'.
```

Seems like this is partially related to https://github.com/goreleaser/goreleaser/pull/4969 . Thanks for catching this @SuperSandro2000